src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -304,10 +329,10 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 					resource: undefined,
 					options: {
 						pinned: true,
-						index: this.group.count, // always at the end
+						index: this.groupViewer.count, // always at the end
```

Comment:
I start to wonder if we might want to allow to create pinned tabs via double click? Might be cool way to quickly have a pinned untitled tab there.